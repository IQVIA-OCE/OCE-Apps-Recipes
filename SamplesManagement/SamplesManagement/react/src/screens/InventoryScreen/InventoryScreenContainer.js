import React from 'react';
import { Platform, StyleSheet } from 'react-native';
import InventoryContext from './InventoryContext';
import InventoryScreenComponent from './InventoryScreenComponent';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import { useTheme } from 'apollo-react-native';

const InventoryScreenContainer = ({ navigation }) => {
  const theme = useTheme();

  return (
    <InventoryContext>
      <KeyboardAwareScrollView
        scrollEnabled={true}
        automaticallyAdjustContentInsets={false}
        keyboardShouldPersistTaps="always"
        scrollEventThrottle={10}
        contentContainerStyle={styles.root}
        style={{ backgroundColor: theme.colors.background, height:  Platform.OS === 'web' ? 600 : '100%' }}
        bounces={false}
      >
        <InventoryScreenComponent navigation={navigation} />
      </KeyboardAwareScrollView>
    </InventoryContext>
  );
};

const styles = StyleSheet.create({
  root: {
    flexGrow: 1,
    height: '100%',
  },
});

export default InventoryScreenContainer;
