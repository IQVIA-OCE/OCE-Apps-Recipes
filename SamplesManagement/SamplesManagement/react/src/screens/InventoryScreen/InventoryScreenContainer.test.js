import React from 'react';
import {render, waitFor} from '@testing-library/react-native';
import InventoryScreenContainer from './InventoryScreenContainer';
import { customSettings } from 'oce-apps-bridges';
import {
  fetchActiveLotsProducts,
  fetchInventoryTypes,
  fetchLastSubmittedInventory,
  fetchTransactionDetails
} from '../../api/Inventories';

jest.mock('../../api/Inventories');

jest.mock('oce-apps-bridges');

jest.mock('react-native-keyboard-aware-scroll-view', () => {
  const KeyboardAwareScrollView = ({ children }) => children;
  return { KeyboardAwareScrollView };
});

jest.mock('react-navigation', () => ({
  withNavigation: Component => props => (
    <Component navigation={{ navigate: jest.fn(), state: { params: { recordType: 'AddHocInventory' } } }} {...props} />
  ),
}));

describe('InventoryScreenContainer', () => {
  beforeAll(() => {
    fetchInventoryTypes.mockResolvedValue([[{
      Id: '1',
      DeveloperName: 'AdHocInventory'
    }]]);

    fetchLastSubmittedInventory.mockResolvedValue([{
      Id: '0',
      createdDate: '123'
    }]);

    fetchTransactionDetails.mockResolvedValue([[]]);
    fetchActiveLotsProducts.mockResolvedValue([[]]);
  })
  it('should render properly', async () => {


    customSettings.getCustomSetting.mockReturnValue({});

    const { getByText} = render(<InventoryScreenContainer />)

    await waitFor(() => {
      expect(getByText(/Inventory Date Time/)).toBeTruthy();
    });
  });
});
