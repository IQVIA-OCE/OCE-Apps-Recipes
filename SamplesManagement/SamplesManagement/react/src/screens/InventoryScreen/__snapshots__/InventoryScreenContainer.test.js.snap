// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryScreenContainer should render properly 1`] = `
<withNavigation(ContextProvider)>
  <KeyboardAwareScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Object {
        "flexGrow": 1,
        "height": "100%",
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="always"
    scrollEnabled={true}
    scrollEventThrottle={10}
    style={
      Object {
        "backgroundColor": "#F6F6F6",
      }
    }
    viewIsInsideTabBar={false}
  >
    <InventoryScreenComponent />
  </KeyboardAwareScrollView>
</withNavigation(ContextProvider)>
`;
